go/service/chat_local.go

```diff
@@ -816,11 +805,35 @@ func (h *chatLocalHandler) postAttachmentLocal(ctx context.Context, arg postAtta
 		chatUI.ChatAttachmentUploadProgress(ctx, parg)
 	}
 
-	// get s3 upload params from server
-	params, err := h.remoteClient().GetS3Params(ctx, arg.ConversationID)
+	// Send a placeholder attachment message that will
+	// be edited after the assets are uploaded.  Sending
+	// it now to preserve the order of send messages.
+	placeholder, err := h.postAttachmentPlaceholder(ctx, arg)
 	if err != nil {
-		return chat1.PostLocalRes{}, err
+		return placeholder, err
 	}
+	h.G().Log.Debug("placeholder message id: %v", placeholder.MessageID)
+
+	// if there are any errors going forward, delete the placeholder message
+	defer func() {
+		if err == nil {
+			return
+		}
+
+		h.G().Log.Debug("postAttachmentLocal error after placeholder message sent, deleting placeholder message")
+		deleteArg := chat1.PostDeleteNonblockArg{
+			ConversationID:   arg.ConversationID,
+			IdentifyBehavior: arg.IdentifyBehavior,
+			Conv:             arg.ClientHeader.Conv,
+			Supersedes:       placeholder.MessageID,
+			TlfName:          arg.ClientHeader.TlfName,
+			TlfPublic:        arg.ClientHeader.TlfPublic,
+		}
+		_, derr := h.PostDeleteNonblock(ctx, deleteArg)
```

Comment:
I figured it had the best chance for success.  

If the attachment upload fails because the internet goes down, then putting the delete in the outbox will eventually remove the placeholder message .